scons/site_tools/stm32_parse_headers.py

```diff
@@ -0,0 +1,48 @@
+import re
+import os
+import codecs
+from collections import OrderedDict
+
+def parse_header(env, header_path, output_path):
+	symbols = {}
+
+	device_header = ["".join(header_path[-13:])]
+	with codecs.open(header_path, 'r', 'ISO-8859-1') as deviceFile:
+		deviceFileContent = deviceFile.read()
+		# find all peripherals
+		peripherals = re.findall(r"#define +(.*?) +\(\((.*?)_TypeDef +\*\) +(.*?_BASE)\)", deviceFileContent)
+		peripherals = {p[0]: (p[1], p[2]) for p in peripherals}
+		# reduce search window to only Peripheral_memory_map section
+		memmapContent = str(re.findall(r"Peripheral_memory_map.*?@\}", deviceFileContent, re.DOTALL | re.MULTILINE)[0])
+		# find all memory mappings
+		rest = re.findall(r"#define +(.*?) +\(?(?:\(uint32_t\))?(.*?)U?\)? *?(\n|/)", memmapContent)
```

Comment:
Also: Nice use of `codecs`, didn't know about that, totally gonna steal that ;-P